src/CMakeLists.txt

```diff
@@ -462,6 +466,10 @@ ADD_SUBDIRECTORY ( Commands )
 ADD_SUBDIRECTORY ( Desktop )
 ADD_SUBDIRECTORY ( CommandLine )
 ADD_SUBDIRECTORY ( Tests )
+if (CMAKE_COMPILER_IS_GNUCC)
```

Comment:
I hadn't used these modules on anything else than x86_64 so far but I like that idea. There is one problem in the case of the dot module, though. While dot.c/h can easily be made to work as you suggested (on unsupported arch use naive dot implementation; see https://github.com/kloewe/mod-dot/commit/e226be9) the current CMakeLists.txt file sets up not only dot.c but also dot_naive.c, dot_sse2.c, and dot_avx.c for compilation and linking. In the non-x86_64 case, only dot.c and dot_naive.c are needed, though, and trying to compile the (unneeded) other objects will probably fail, or am I missing something here? Do you know a way around this that doesn't involve having some conditionals in the CMakeLists.txt similar to what we discussed above?

Meanwhile, I have implemented your proposed solution (CPUINFO_COMPILES etc.) for now. I have tried to set it up in such a way that there is only one line in src/CMakeLists.txt that needs to be changed if you wanted to use this for your Apple Clang builds too.